mm2src/coins_activation/src/utxo_activation/init_qtum_activation.rs

```diff
@@ -0,0 +1,95 @@
+use crate::context::CoinsActivationContext;
+use crate::prelude::TryFromCoinProtocol;
+use crate::standalone_coin::InitStandaloneCoinActivationOps;
+use crate::utxo_activation::init_utxo_standard_activation::{UtxoStandardRpcTaskHandle, UtxoStandardTaskManagerShared};
+use crate::utxo_activation::init_utxo_standard_activation_error::InitUtxoStandardError;
+use crate::utxo_activation::init_utxo_standard_statuses::{UtxoStandardAwaitingStatus, UtxoStandardInProgressStatus,
+                                                          UtxoStandardUserAction};
+use crate::utxo_activation::utxo_standard_activation_result::UtxoStandardActivationResult;
+use async_trait::async_trait;
+use coins::hd_pubkey::RpcTaskXPubExtractor;
+use coins::utxo::qtum::{QtumCoin, QtumCoinBuilder};
+use coins::utxo::utxo_builder::UtxoCoinBuilder;
+use coins::utxo::UtxoActivationParams;
+use coins::{lp_register_coin, CoinProtocol, MmCoinEnum, PrivKeyBuildPolicy, RegisterCoinParams};
+use common::mm_ctx::MmArc;
+use common::mm_error::prelude::*;
+use crypto::hw_rpc_task::HwConnectStatuses;
+use serde_json::Value as Json;
+
+pub struct QtumProtocolInfo;
+
+impl TryFromCoinProtocol for QtumProtocolInfo {
+    fn try_from_coin_protocol(proto: CoinProtocol) -> Result<Self, MmError<CoinProtocol>>
+    where
+        Self: Sized,
+    {
+        match proto {
+            CoinProtocol::QTUM => Ok(QtumProtocolInfo),
+            protocol => MmError::err(protocol),
+        }
+    }
+}
+
+#[async_trait]
+impl InitStandaloneCoinActivationOps for QtumCoin {
+    type ActivationRequest = UtxoActivationParams;
+    type StandaloneProtocol = QtumProtocolInfo;
+    type ActivationResult = UtxoStandardActivationResult;
+    type ActivationError = InitUtxoStandardError;
+    type InProgressStatus = UtxoStandardInProgressStatus;
+    type AwaitingStatus = UtxoStandardAwaitingStatus;
+    type UserAction = UtxoStandardUserAction;
+
+    fn rpc_task_manager(activation_ctx: &CoinsActivationContext) -> &UtxoStandardTaskManagerShared {
+        &activation_ctx.init_utxo_standard_task_manager
+    }
+
+    async fn init_standalone_coin(
+        ctx: MmArc,
+        ticker: String,
+        coin_conf: Json,
+        activation_request: Self::ActivationRequest,
+        _protocol_info: Self::StandaloneProtocol,
+        priv_key_policy: PrivKeyBuildPolicy<'_>,
+        task_handle: &UtxoStandardRpcTaskHandle,
+    ) -> Result<Self, MmError<Self::ActivationError>> {
+        // Construct an Xpub extractor without checking if the MarketMaker supports HD wallet ops.
+        // If the coin builder tries to extract an extended public key despite HD wallet is not supported,
+        // [`UtxoCoinBuilder::build`] fails with the [`UtxoCoinBuildError::IguanaPrivKeyNotAllowed`] error.
+        let xpub_extractor = RpcTaskXPubExtractor::new_unchecked(&ctx, task_handle, HwConnectStatuses {
```

Comment:
It seems to me that `HwConnectStatuses` can be simplified. I can see some similarities in how it is created in various contexts.

Reply:
I've added [coins_activation::utxo_activation::common_impl::xpub_extractor_rpc_statuses](https://github.com/KomodoPlatform/atomicDEX-API/blob/init-utxo-with-trezor/mm2src/coins_activation/src/utxo_activation/common_impl.rs#L48)

Reply:
Ok, thanks.